client/asset/eth/node.go

```diff
@@ -32,8 +33,68 @@ var simnetGenesis string
 type nodeConfig struct {
 	net                dex.Network
 	listenAddr, appDir string
+	logger             dex.Logger
 }
 
+// ethLogger satisifies geth's logger interface.
+type ethLogger struct {
+	dl dex.Logger
+}
+
+// New returns a new Logger that has this logger's context plus the given
+// context.
+func (el *ethLogger) New(ctx ...interface{}) log.Logger {
+	s := ""
+	for _, v := range ctx {
+		s = fmt.Sprintf("%s:%s", s, v)
+	}
+	l := el.dl.SubLogger(s)
+	return &ethLogger{dl: l}
+}
+
+// GetHandler gets the handler associated with the logger. Unused in geth. Does
+// nothing and returns a nil interface here.
+func (el *ethLogger) GetHandler() log.Handler {
+	return nil
+}
+
+// SetHandler updates the logger to write records to the specified handler.
+// Used during setup in geth when a logger is not supplied. Does nothing here.
+func (el *ethLogger) SetHandler(h log.Handler) {}
+
+// Trace logs at Trace level.
+func (el *ethLogger) Trace(msg string, ctx ...interface{}) {
+	el.dl.Trace(append([]interface{}{msg}, ctx...))
```

Comment:
Why not `el.dl.Tracef(msg, ctx...)`?

Reply:
Oh, I just looked at the go-ethereum code and I see that `ctx` is supposed to be key/value pairs, with the entirety of the formatted message in `mgs`.  e.g. `log.Debug("Failed to decode keystore key", "path", path, "err", err)`  How should that display?

Reply:
How would you like it to display? lol. I can't really see any different using this and the default logger. will get a comparison here in a sec...

Reply:
Yeah, it's totally up to us.  I'm just wondering how geth (the app) would print that out, so I can understand what's usually in these key-value pairs in `ctx`.  I grabbed that line randomly from the go-ethereum code base.  Seems like the intention is to show something like:

```
[DBG] GETH: Failed to decode keystore key. Details: path = /mnt/blah/key.db, err = "oh darn"
```

Reply:
them
```
DEBUG[07-27|15:20:52.909] Removing p2p peer                        peercount=2 id=bad85b887d6606b2 duration="355.116µs" req=false err="Key missing from list - recentTxLookup"
```
us
```
2021-07-26 17:53:35.039 [TRC] ETHTEST[NODE]2021-07-26 17:53:35.039 [DBG] ETHTEST[NODE]: [Removing p2p peer peercount 2 id bad85b887d6606b246bb493caa638499f2f0576a7c2ed84fc72706b3de7ddceb duration 446.036µs req false err Key missing from list - recentTxLookup]
```
I'll make it more like theirs.

Reply:
at latest commit:
```
2021-07-27 16:33:56.327 [DBG] ETHTEST[NODE]: Removing p2p peer             peercount = 3   id = a150cfb3940507cc8b340fb6b647855bf674ee82afec58f0c8426a5f354d2107   duration = 568.345µs   req = false   err = Key missing from list - recentTxLookup
```

Reply:
Works for me.  A little space-heavy, but OK.